core/src/main/scala/io/projectglow/sql/expressions/FirthTest.scala

```diff
@@ -85,8 +90,8 @@ object FirthTest extends LogitTest {
 
     while (!converged && !exploded && iter <= maxIter) {
       try {
-        args.mu := x(::, 0 until m0) * args.b
-        sigmoid.inPlace(args.mu)
+        val eta = offsetOption.fold(x(::, 0 until m0) * args.b)(_ + x(::, 0 until m0) * args.b)
```

Comment:
Prefer pattern matching or if else over fold as per https://github.com/databricks/scala-style-guide#monadic-chaining

Reply:
changed to pattern matching everywhere.